process/block/shardblockProposal.go

```diff
@@ -366,3 +495,36 @@ func (sp *shardProcessor) selectOutgoingTransactions() [][]byte {
 
 	return outgoingTransactions
 }
+
+func (sp *shardProcessor) checkMetaHeadersValidityAndFinalityProposal(header data.ShardHeaderHandler) error {
+	lastCrossNotarizedHeader, _, err := sp.blockTracker.GetLastCrossNotarizedHeader(core.MetachainShardId)
+	if err != nil {
+		return err
+	}
+	usedMetaHdrHashes := header.GetMetaBlockHashes()
+	usedMetaHeaders := make([]data.HeaderHandler, 0, len(usedMetaHdrHashes))
+	for _, metaHdrHash := range usedMetaHdrHashes {
+		metaHdr, errNotCritical := sp.dataPool.Headers().GetHeaderByHash(metaHdrHash)
```

Comment:
`errNotCritical` -> `err`

expecting `errNotCritical` to result in a continue or log only 🤔